ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/Situation.java

```diff
@@ -0,0 +1,299 @@
+     *
+     * @param variantId1: First variant to compare.
+     * @param variantId2: Second variant to compare.
+     * @return True if all the range actions are set at the same values and false otherwise.
+     */
+    public boolean sameRemedialActions(String variantId1, String variantId2) {
+        String preventiveState = getCrac().getPreventiveState().getId();
+        for (RangeAction rangeAction : getCrac().getRangeActions()) {
+            RangeActionResultExtension rangeActionResultMap = rangeAction.getExtension(RangeActionResultExtension.class);
+            double value1 = rangeActionResultMap.getVariant(variantId1).getSetPoint(preventiveState);
+            double value2 = rangeActionResultMap.getVariant(variantId2).getSetPoint(preventiveState);
+            if (value1 != value2 && (!Double.isNaN(value1) || !Double.isNaN(value2))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    // VARIANTS MANAGEMENT METHODS
+
+    /**
+     * This method generates a unique variant ID of a network according to the IDs already present in the network.
+     *
+     * @return A unique ID of the network as a string.
+     */
+    private String getUniqueNetworkVariantId() {
+        String uniqueId;
+        do {
+            uniqueId = UUID.randomUUID().toString();
+        } while (network.getVariantManager().getVariantIds().contains(uniqueId));
+        return uniqueId;
+    }
+
+    /**
+     * This method generates a unique variant ID of a situation. This ID has to be unique for the network
+     * and for the CRAC.
+     *
+     * @return A unique ID a the situation as a string.
+     */
+    private String getUniqueSituationId() {
+        String networkVariantId = getUniqueNetworkVariantId();
+        String cracVariantId = crac.getExtension(ResultVariantManager.class).getUniqueVariantId();
+        return networkVariantId.concat(cracVariantId);
+    }
+
+    private String createVariantFromNetworkVariant(String networkVariantId) {
+        String situationVariantId = getUniqueSituationId();
+        network.getVariantManager().cloneVariant(networkVariantId, situationVariantId);
+        crac.getExtension(ResultVariantManager.class).createVariant(situationVariantId);
+        variantIds.add(situationVariantId);
+        systematicSensitivityAnalysisResultMap.put(situationVariantId, null);
+        return situationVariantId;
+    }
+
+    /**
+     * This methods creates a variant from the network working variant. It is useful for the constructor or
+     * after a clear method that will erase all the situation variants. Except these two cases it is advised
+     * to use cloneVariant method instead. Network variant will be copied from its working variant,
+     * CRAC result variant will be created but empty as well as sensitivity computation results variant.
+     *
+     * @return The situation variant ID of the newly created variant.
+     */
+    String createVariant() {
```

Comment:
delete this method. Small and only for internal uses. We don't want it to be used outside of this class